src/Ocelot/Authorization/ClaimsAuthorizer.cs

```diff
@@ -32,7 +32,7 @@ List<PlaceholderNameAndValue> urlPathPlaceholderNameAndValues
                 if (values.Data != null)
                 {
                     // dynamic claim
-                    var match = Regex.Match(required.Value, @"^{(?<variable>.+)}$");
+                    var match = Regex.Match(required.Value!.FirstOrDefault() ?? string.Empty, "^{(?<variable>.+)}$");
```

Comment:
That's not your code, I agree, Maybe we could use a compiled Regex and enhance security by specifying a timeout, 10 seconds...
```c#
[GeneratedRegex("^{(?<variable>.+)}$", RegexOptions.Compiled, 10000)]
private static partial Regex DynamicClaimRegex();
```

Reply:
in GenerateRegex, RegexOptions.Compiled is ignored

Reply:
So `RegexOptions.Compiled` makes sense for static variable, right?
We have well known open PR by Mohsen who followed `Regex` best practices, see #1348